llvm/lib/Target/RISCV/RISCVSchedSiFive7.td

```diff
@@ -455,11 +455,20 @@ foreach mx = SchedMxList in {
 // specific suffixes, but since SEW is already encoded in the name of the
 // resource, we do not need to use LMULSEWXXX constructors. However, we do
 // use the SEW from the name to determine the number of Cycles.
+
+// This predicate is true when the rs2 operand of vlse or vsse is x0, false
+// otherwise.
+def VLDSX0Pred: SchedPredicate<[{ MI->getOperand(3).getReg() == RISCV::X0 }]>;
```

Comment:
Can this be rewritten by `CheckRegOperand` defined in `TargetInstrPredicate.td`?